yt/frontends/amrvac/data_structures.py

```diff
@@ -245,56 +245,50 @@ def _set_code_unit_attributes(self):
+            velocity_override = self.units_override.get('velocity_unit', (0, 'cm*s**-1'))
+            temperature_override = self.units_override.get('temperature_unit', (1, 'K'))
+            mylog.info('Overriding numberdensity_unit: {:1.0e} {}.'.format(numberdensity_override[0],
+                                                                           numberdensity_override[1]))
         else:
-            # Next, check for unit normalisations in the .dat file. Do not use .get() with default value added
-            # to explicitly print a warning to the console.
-            try:
-                unit_length = (self.parameters['unit_length'], 'cm')
-                unit_numberdensity = (self.parameters['unit_numberdensity'], 'cm**-3')
-                unit_temperature = (self.parameters['unit_temperature'], 'K')
-                unit_velocity = (self.parameters['unit_velocity'], 'cm*s**-1')
-            # If nothing present in datfile, use default values and print warning
-            except KeyError:
-                mylog.warning('No unit normalisations found in datfile and no overrides specified. Using defaults.')
-                unit_length = (1, 'cm')
-                unit_numberdensity = (1, 'cm**-3')
-                unit_temperature = (1, 'K')
-                unit_velocity = (0, 'cm*s**-1')
-
-        # Convert to YT quantities
-        unit_length = self.quan(unit_length[0], unit_length[1])
-        unit_numberdensity = self.quan(unit_numberdensity[0], unit_numberdensity[1])
-        unit_temperature = self.quan(unit_temperature[0], unit_temperature[1])
-        unit_velocity = self.quan(unit_velocity[0], unit_velocity[1])
-
-        He_abundance = 0.1  # hardcoded in AMRVAC
-
-        unit_density = (1.0 + 4.0*He_abundance) * mass_hydrogen_cgs * unit_numberdensity
-        if unit_velocity == 0:
-            unit_pressure = ((2.0 + 3.0*He_abundance) *
-                             unit_numberdensity * boltzmann_constant_cgs  * unit_temperature).to('dyn*cm**-2')
-            unit_velocity = (np.sqrt(unit_pressure / unit_density)).to('cm*s**-1')
+            # if nothing is specified, use the default AMRVAC normalisations
+            length_override = (1, 'cm')
+            numberdensity_override = (1, 'cm**-3')
+            temperature_override = (1, 'K')
+            velocity_override = (0, 'cm*s**-1')
+
+        # Create YT quantities
+        length_unit = self.quan(length_override[0], length_override[1])
+        numberdensity_unit = self.quan(numberdensity_override[0], numberdensity_override[1])
+        temperature_unit = self.quan(temperature_override[0], temperature_override[1])
+        velocity_unit = self.quan(velocity_override[0], velocity_override[1])
+
+        He_abundance = 0.1  # hardcoded parameter in AMRVAC
+        density_unit = (1.0 + 4.0*He_abundance) * mass_hydrogen_cgs * numberdensity_unit
+        if velocity_unit == 0:
```

Comment:
still doubt this comparison very much. It may work (depending if the comparison operator is defined between units and integers) but it's sloppy